sdk/trace/sampling.go

```diff
@@ -30,13 +31,12 @@ type Sampler interface {
 
 // SamplingParameters contains the values passed to a Sampler.
 type SamplingParameters struct {
-	ParentContext   trace.SpanContext
-	TraceID         trace.TraceID
-	Name            string
-	HasRemoteParent bool
-	Kind            trace.SpanKind
-	Attributes      []attribute.KeyValue
-	Links           []trace.Link
+	ParentContext context.Context
```

Comment:
Why do we need the `context.Context` here, it seems that every usage of `ParentContext` is to get the `SpanContext` out of it.

Shouldn't we use `SpanContext` directly?

Reply:
Agreed, it's not the best so far because we only ever use the `SpanContext`. However, the [specification requires](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.0.1/specification/trace/sdk.md#sampler) this parameter to be this:

> * [`Context`](../context/context.md) with parent `Span`.
>   The Span's SpanContext may be invalid to indicate a root span.
> ...

Additionally, this means that if a sampler wanted to perform tasks for things not contained in a `SpanContext` (e.g. attributes, links, events) it will be able to use this information.

Reply:
Gotcha, so the default samplers doesn't use it but it should be open for extensions.